org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/editor/folding/FoldingRange.java

```diff
@@ -0,0 +1,55 @@
+/*******************************************************************************
+ * Copyright (c) 2021 TypeFox GmbH (http://www.typefox.io) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.xtext.ide.editor.folding;
+
+/**
+ * @author Mark Sujew - Initial contribution and API
+ * 
+ * @since 2.25
+ */
+public class FoldingRange {
+
+	private int offset;
```

Comment:
We collect the foldings in a HashSet by default - should this implement hashCode and equals?